txnkv/transaction/2pc.go

```diff
@@ -1222,6 +1280,16 @@ func (c *twoPhaseCommitter) execute(ctx context.Context) (err error) {
 	err = c.prewriteMutations(bo, c.mutations)
 
 	if err != nil {
+		if strings.Contains(err.Error(), "assertion fail") {
+			ts, err1 := c.store.GetTimestampWithRetry(retry.NewBackofferWithVars(ctx, TsoMaxBackoff, c.txn.vars), c.txn.GetScope())
+			if err1 == nil {
+				_, _, err2 := c.checkSchemaValid(ctx, ts, c.txn.schemaVer, false)
+				if err2 != nil {
+					err = err2
+				}
+			}
```

Comment:
I guess it is to avoid some false positives with online DDL? Could you add some comments?